### PR TITLE
Implement createPoints.py with geopandas

### PR DESCRIPTION
**Pull request by jayqi**
This is a rewrite of [`createPoints.py`](https://github.com/mittrees/Treepedia_Public/blob/master/Treepedia/createPoints.py) from [mittrees/Treepedia_Public](https://github.com/mittrees/Treepedia_Public/) using [geopandas](https://geopandas.org/en/stable/), a modern popular geospatial Python library. 

A few key points:

- This version uses geopandas instead of loading the data directly with fiona. Using geopandas is a generally nicer and easier developer experience than directly loading vector features with fiona and manipulating them in loops.
- This version is more flexible than `createPoints.py` in accepting and writing different vector data formats.
- This version is more flexible than `createPoints.py` in accepting input data in any CRS as long as the CRS is defined in the data itself.
- This version also splits out the processing steps into functions for more modular reuse as a library. 

You can see the Jupyter notebook `notebooks/0.0-jyq-create-points-demo.ipynb` for a demo. Note that the folium maps do not directly load in GitHub, but you can see the rendered notebook [here via nbviewer](https://nbviewer.org/github/jayqi/street-view-green-view/blob/jyq-create-points-geopandas/notebooks/0.0-jyq-create-points-demo.ipynb).

The script can also be called as a command-line program with `python src/create_points.py` or `python -m src.create_points`. Use the `--help` flag for documentation.

### CODE REVIEW

**Comment by jayqi** [comment]
@danbjoseph It looks like that road has the property `highway=tertiary`. It's one of the road types that gets filtered out: 

https://github.com/jayqi/street-view-green-view/blob/c2228ab5c4b217db9f0f8b189669541ed477cece/src/create_points.py#L33

This is logic from the Treepedia version:

https://github.com/mittrees/Treepedia_Public/blob/34d925c1f85e44c061a3ecae5ce93e6bd4d94c2a/Treepedia/createPoints.py#L35

https://github.com/mittrees/Treepedia_Public/blob/34d925c1f85e44c061a3ecae5ce93e6bd4d94c2a/Treepedia/createPoints.py#L49-L61

File: src/create_points.py
Comment (ayoakin) [inline src/create_points.py:1]: Hi @jayqi Thanks for this!

I tried to run this but keep getting `'highway' column not found in input GeoDataFrame. Input data must be of OpenStreetMap roads.` for the sample files I used.
Did you run with sample files? What files did you use?

cc: @danbjoseph

Comment (jayqi) [reply src/create_points.py:1]: I tested against `Three_Rivers_Michigan_USA_shp_uid_ae45a8ad-f097-4fd3-bec9-31a366f92657 2.zip` from the shared Google Drive, which is the ZIP archive that has shapefiles in it.

**Comment by jayqi** [comment]
@ayoakin did you mean to merge or was that a mistake? I didn't think @danbjoseph had finished reviewing.

**Comment by danbjoseph** [comment]
i'm not terribly concerned about what's in a given PR at this point (no one outside those working on the project is using the code). i asked @ayoakin to review as he had been building familiarity with the code while working in a Google Colab. 

@jayqi please note my comment about the final output from this step being a points file (not lines). the other items can come later (e.g. figuring out the best method for cleaning/filtering the different roads features... should the user do it in a GIS software before starting the scripted process? should we have configurable filters in the code? something else?) as they aren't blocking moving onto the Mapillary API calls and other steps

**Comment by ayoakin** [comment]
@jayqi I merged so I can pull and check as we don't have a job for that atm on here. Like @danbjoseph mentioned I don't think what's in main should matter at this point.

I am mainly checking for functionality and reproducibility. Might make some suggestions once I am able to make it run locally on my end. Thanks so much!

**Comment by jayqi** [comment]
> @jayqi please note my comment about the final output from this step being a points file (not lines). the other items can come later (e.g. figuring out the best method for cleaning/filtering the different roads features... should the user do it in a GIS software before starting the scripted process? should we have configurable filters in the code? something else?) as they aren't blocking moving onto the Mapillary API calls and other steps

The output is indeed a file of points, and sorry that this wasn't clear. In the notebook, I've subsetted the map to only show a single road for as a demonstration, but the actual workflow of using the script produces a output file that contains the interpolated points for all roads (with the same filtering in the Treepedia script).

**Pull request merged**